fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaDataBuilder.java

```diff
@@ -535,6 +575,11 @@ private static void validateRecords(@Nonnull Descriptors.FileDescriptor fileDesc
         return union;
     }
 
+    private boolean isImported(@Nonnull Descriptors.Descriptor descriptor) {
+        return !recordsDescriptor.getMessageTypes().stream()
+                .map(Descriptors.Descriptor::getName).filter(descriptor.getName()::equals).findFirst().isPresent();
```

Comment:
A previous version checked the message descriptor for pointer equality with one of the two file descriptors (either the local one or the one from the proto). I thought that was sufficient; why was this changed?

Reply:
Because it breaks when a record type used to be imported but it's moved to the local file descriptor. Hence, the record type will not be a message type in recordsDescriptor but since it is a message type in the local descriptor, the files will match.

Reply:
done.